v2/call_option.go

```diff
@@ -63,6 +63,52 @@ func WithRetry(fn func() Retryer) CallOption {
 	return retryerOption(fn)
 }
 
+// OnErrors returns a Retryer that retries if and only if the previous attempt
+// returns an error that matches one of the errors listed in errs when compared
+// with the given compare function. Supplying errors.Is (https://pkg.go.dev/errors#Is)
+// for compare is valid.
+//
+// Pause times between retries are specified by bo. bo is only used for its
+// parameters; each Retryer has its own copy.
+func OnErrors(bo Backoff, compare func(err, target error) bool, errs ...error) Retryer {
```

Comment:
I know you mentioned this in the PR comment, but I too am not really sure if this is needed.

With HTTP codes as an example you could simply case to the googleapi.Error and compare against a list of codes in /w the other methods.

Reply:
You are right, this isn't strictly necessary with `OnError` being there. But I think this provides a usability improvement when you have multiple functions using the same _heuristic_ (i.e. comparing `googleapi.Error.Code`) but a different set of `error`s to retry on (e.g. one func retries on 503 and 502 and another only retries on 503). One would. need a different `shouldRetry` function for each different set of `error`s. This would be the case for GAPICs certainly. There is a long discussion about this in our chat thread with @tbpg.

Reply:
I actually think this is a bit clunky for something like HTTP errors, no? Thinking about the Go storage library, I would have to leak some convoluted logic into the comp function to cover all my cases (because I can get a variety of error types)-- I think would have to get into casting/interfaces for BOTH err and target. The `shouldRetry` surface seems more straightforward honestly.

Reply:
@tritone Fair enough! I'm convinced :)